pkg/database/realm.go

```diff
@@ -469,14 +469,14 @@ func (r *Realm) GetLongCodeDurationHours() int {
 	return int(r.LongCodeDuration.Duration.Hours())
 }
 
-// EffectiveMFAMode returns the realm's default MFAMode but first
-// checks if the user is in the grace-period (if so, required becomes prompt).
-func (r *Realm) EffectiveMFAMode(user *User) AuthRequirement {
+// EffectiveMFAMode returns the realm's default MFAMode but first checks if the
+// time is in the grace-period (if so, required becomes prompt).
+func (r *Realm) EffectiveMFAMode(t time.Time) AuthRequirement {
```

Comment:
alternative: pass in the membership, use membership.CreatedAt if not null or fall back to user.CreatedAt?

Reply:
actually looks like we make that not-null and set to now(). I wonder will that be a problem for existing users (gives them an extra N days)?

Reply:
meh this is just for grace periods. I've convinced myself this is fine